### PR TITLE
Added a new user type,  created a shopping cart with a client, applied discount strategy pattern design

### PR DESCRIPTION
**Pull request by sabiutayo**
Added a new user type,
created a shopping cart with a client,
applied discount strategy pattern design

### CODE REVIEW

**Review by Bossy12 — APPROVED** [review]

File: src/main/java/sample/main2.java [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 
 public class main2 {
     public static void main(String[] args) {
         User user = new User();
-        user.setUsername("Marcelino");
-        user.setPassword("password");
-        user.setUserType(UserType.CUSTOMER);
+        user.setUsername("Norocosul");
+        user.setPassword("000000");
+//        user.setUserType(UserType.CUSTOMER);
```
Comment (stefan2212) [inline src/main/java/sample/main2.java]: Usually, commented code is not good. You have versioning now. If you need something that you deleted you can always recover it

**Review by stefan2212 — CHANGES_REQUESTED** [review]

File: src/main/java/sample/main2.java (hunk 1 above)
Comment (sabiutayo) [reply src/main/java/sample/main2.java]: This class is just for test and debug.

**Review by MariusTarnaru — APPROVED** [review]
What is the reason to create usertype LUCKYONE ?

**Comment by sabiutayo** [comment]
> What is the reason to create usertype LUCKYONE ?

In case we have a promotion on our market place (like : 1st 100 costumer, i don't know)

File: src/main/java/sample/Utils/UserDao.java [hunk 2]
```diff
@@ -3,32 +3,28 @@
...
 import java.util.List;
 
 public class UserDao {
     private Session session = HibernateUtil.getSession();
 
-    public List getAllUsers() {
-        List<User> users = session.createQuery("from User").list();
-        return users;
+    public List<User> getAllUsers() {
+        return session.createQuery("from User").list();
```
Comment (stefan2212) [inline src/main/java/sample/Utils/UserDao.java]: This one can be ambiguous. Try completing the query

**Review by stefan2212 — CHANGES_REQUESTED** [review]

**Review by MariusTarnaru — APPROVED** [review]

File: src/main/java/sample/Utils/CustomerDiscountClient.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    public double applyDiscount(double totalToPay) {
+        return discount * totalToPay;
+    }
+
+    public double getDiscount() {
+        return discount;
+    }
+
+    }
+
```
Comment (stefan2212) [inline src/main/java/sample/Utils/CustomerDiscountClient.java]: Too many empty lines.

File: src/main/java/sample/Utils/LuckyOneDiscountClient.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package sample.Utils;
+
+public class LuckyOneDiscountClient implements DiscountClientStrategy {
```
Comment (stefan2212) [inline src/main/java/sample/Utils/LuckyOneDiscountClient.java]: Let's name the Interface Client and from the implementations let's remove the Discount part

**Review by stefan2212 — CHANGES_REQUESTED** [review]

**Review by Bossy12 — APPROVED** [review]

**Review by MariusTarnaru — APPROVED** [review]

**Review by stefan2212 — APPROVED** [review]

**Pull request merged**